frontend/packages/sdk/src/OnChainRegistry.ts

```diff
@@ -200,18 +201,26 @@ export class OnChainRegistry {
     const result = await this.api.query.phalaPhatContracts.clusterWorkers(_clusterId)
     const workerIds = result.toJSON() as string[]
     const infos = await this.api.query.phalaRegistry.endpoints.multi(workerIds)
-    return infos.map((info, idx) => {
+    let idx = 0
+    return infos.reduce((prev, info) => {
```

Comment:
Can use `infos.filter(..... => (info as Option<Codec>).isSome).map(....)` for better readability

Reply:
The result is not the same because we need a consistent `idx` to access the corresponding `workerId`, check the updated version.

Reply:
We can map first, then filter. This avoids using `...prev` and creating a new idx variable inside the scope.
```ts
return infos.map((info, idx) => {
  const workerId = workerIds[idx]
  const { v1 } = info.toJSON()

  if (v1 && v1.length) {
    return {
      pubkey: workerId,
      clusterId: _clusterId,
      endpoints: {
        default: v1[0],
        v1
      }
    }
  }
  return null
}).filter(Boolean)

```